code/scripting/lua/LuaValue.h

```diff
@@ -159,6 +159,21 @@ class LuaValue {
 		}
 	}
 
+	/**
+	 * @brief Gets the value of a userdata type using the passed getter function or throws an exception
+	 *
+	 * @exception LuaException Thrown when the conversion failed.
+	 */
+	template<typename Type>
+	void getValue(scripting::ade_odata_getter<Type>&& od) const {
+		_reference->pushValue(_luaState);
+
+		if (!convert::popValue<Type>(_luaState, std::forward<scripting::ade_odata_getter<Type>>(od))) {
+			lua_pop(_luaState, 1);
+			throw LuaException("Failed to pop value");
+		}
+	}
```

Comment:
This type is needlessly specific. It should be possible to change this to a generic rvalue reference which will avoid the direct dependency on the ade types:
```suggestion
	template<typename Type>
	void getValue(Type&& value) const {
		_reference->pushValue(_luaState);

		if (!convert::popValue<Type>(_luaState, std::forward<Type>(value))) {
			lua_pop(_luaState, 1);
			throw LuaException("Failed to pop value");
		}
	}
```